sqlgen/reflect.go

```diff
@@ -258,17 +203,9 @@ func (s *Schema) buildDescriptor(table string, primaryKeyType PrimaryKeyType, ty
 			return nil, fmt.Errorf("bad type %s: duplicate column %s", typ, column)
 		}
 
-		scalarType := field.Type
-		if field.Type.Kind() == reflect.Ptr {
-			scalarType = field.Type.Elem()
-		}
-
-		scannable, ok := s.scalarTypes[scalarType]
-		if !ok {
-			scannable, ok = coerceToScannable(scalarType)
-		}
-		if !ok {
-			return nil, fmt.Errorf("bad type %s: field %s has unsupported type %s", typ, field.Name, field.Type)
+		d := fields.New(field.Type, tags[1:])
```

Comment:
is it safe for us to reuse the `sql` tag in this manner? should we use a different tag?

Reply:
The sql tag is explicitly being used by only us in sqlgen. I think we should just use `sql` everywhere.

Reply:
ah, yep, that makes sense. i had it in my head that other software used that tag too